### ISSUE TITLE
error output in el7: "date: invalid date ‘+%s’"

### ISSUE THREAD

**Issue by lorengordon** [plus3it/ash-linux-formula#204]
While testing the PR adding validation of salt syntax, I noticed that RHEL-07-020250 is outputting the error, `output: date: invalid date '+%s'`. Appears to be resulting from this command:

```
date -d $(yum history 2> /dev/null | awk -F '|' '$4 ~ / U/{print $3}' | head -1 | sed -e 's/^ *//' -e 's/ .*$//') +%s
```

I'm not super sure what that command is supposed to be doing, but it fails even when run by itself (outside of salt).

```
# date -d $(yum history 2> /dev/null | awk -F '|' '$4 ~ / U/{print $3}' | head -1 | sed -e 's/^ *//' -e 's/ .*$//') +%s
date: invalid date ‘+%s’
```

**Comment by ferricoxide** [comment]
It takes the most recent entry in yum's history and converts it to seconds since epoch (but only if the output from the `awk`-filtered `yum history` output is non-null). Contrast the target output (`date -d ...`) with the unformatted output (`echo ...`):

![image](https://cloud.githubusercontent.com/assets/7087031/21626885/123ffc42-d1e2-11e6-9be3-0ce4d9be67c3.png)

That said, if the test-system has never had updates applied, (what the `awk` checks for), this evaluates to null. Net result, a null-value is passed to the `date -d +%s` conversion to seconds-since-epoch. This, in turn, creates the unhelpful error output.

At any rate, I probably need to update the assumption around the evaluation to account for a "never been updated" system. :-\

**Comment by lorengordon** [comment]
Fixed by #250
